presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -844,13 +846,26 @@ public boolean supportsMissingColumnsOnInsert(Session session, TableHandle table
     }
 
     @Override
-    public InsertTableHandle beginRefreshMaterializedView(Session session, TableHandle tableHandle)
+    public InsertTableHandle beginRefreshMaterializedView(Session session, TableHandle tableHandle, List<TableHandle> sourceTableHandles)
     {
         CatalogName catalogName = tableHandle.getCatalogName();
         CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogName);
         ConnectorMetadata metadata = catalogMetadata.getMetadata();
         ConnectorTransactionHandle transactionHandle = catalogMetadata.getTransactionHandleFor(catalogName);
-        ConnectorInsertTableHandle handle = metadata.beginRefreshMaterializedView(session.toConnectorSession(catalogName), tableHandle.getConnectorHandle());
+
+        List<ConnectorTableHandle> sourceConnectorHandles = sourceTableHandles.stream()
```

Comment:
Should this also verify that source connector handles are for the same connector as the materialized view?

Reply:
Yes.